launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -260,15 +260,14 @@ static String quoteForBatchScript(String arg) {
         quoted.append('"');
         break;
 
-      case '=':
-        quoted.append('^');
-        break;
-
       default:
         break;
       }
       quoted.appendCodePoint(cp);
     }
+    if (arg.codePointAt(arg.length() - 1) == '\\') {
+        quoted.append("\\");
```

Comment:
This is indented too far. Why would a backslash need escaping only in the final position?